Fixed #30913 -- Added support for covering indexes on PostgreSQL 11+. (#11991)
django/django: docs/ref/models/indexes.txt

    The ``condition`` argument is ignored with MySQL and MariaDB as neither
    supports conditional indexes.

``include``
-----------

.. attribute:: Index.include

.. versionadded:: 3.2

> A list or tuple of the name of the fields to be included in the covering index

Reviewer: ```suggestion
A list or tuple of the names of the fields to be included in the covering index
```

as non-key columns. This allows index-only scans to be used for queries that
select only included fields (:attr:`~Index.include`) and filter only by indexed
fields (:attr:`~Index.fields`).

For example::

    Index(name='covering_index', fields=['headline'], include=['pub_date'])

> will allow filtering on ``headline`` and also selecting ``pub_date`` with
> having to only fetch data from the index.

Reviewer: ```suggestion
will allow filtering on ``headline``, also selecting ``pub_date``, while
fetching data only from the index.
```

    The ``condition`` argument is ignored with MySQL and MariaDB as neither
    supports conditional indexes.

``include``

> -------------

Reviewer 2: Please fix the heading underlines to be the same length as the heading itself.

.. attribute:: Index.include

.. versionadded:: 3.1

Used to create covering indexes where all columns in a query is available in

> the index. This reduces the number I/O operations needed to run a query.

Author: I've tried to make it a bit clearer.

For example, ``Index(name='covering_index', fields=['last_name'],

> include=['first_name'])`` will allow filtering on ```last_name`` while also

Reviewer 2: Fix triple backtick to double backtick...

selecting ``first_name`` and only fetch data from the index.

By using :meth:`~django.db.models.query.QuerySet.only()` to select
fields included in the index while also only using these fields in
:meth:`~django.db.models.query.QuerySet.filter()` a fully index covered
query can be produced.

Using ``include`` will produce a smaller index than using a regular
multi-column index but with the drawback that the columns can not be used as
predicates or for sorting and still have the benefits of regular indexed
columns.

:attr:`Index.name` is required when using ``include``.

> ``include`` are ignored for databases besides PostgreSQL.

Reviewer 2: `are` → `is`.
